zilliqa/src/api/types/eth.rs

```diff
@@ -24,6 +24,63 @@ pub enum HashOrTransaction {
     Transaction(Transaction),
 }
 
+#[derive(Clone, Serialize)]
+pub struct QuorumCertificate {
+    #[serde(serialize_with = "hex")]
+    pub signature: Vec<u8>,
+    pub cosigned: String,
+    #[serde(serialize_with = "hex")]
+    pub view: u64,
+    #[serde(serialize_with = "hex")]
+    pub block_hash: B256,
+}
+
+impl QuorumCertificate {
+    pub fn from_qc(qc: &message::QuorumCertificate) -> Self {
+        Self {
+            signature: qc.signature.to_bytes(),
+            cosigned: qc
+                .cosigned
+                .clone()
+                .iter()
+                .fold(String::new(), |mut acc, bit| {
```

Comment:
I think this implementation basically [already exists](https://docs.rs/bitvec/latest/src/bitvec/vec/traits.rs.html#351-365) in the `Display` impl of `BitVec`. Can we instead do
```
#[serde(serialize_with = "ser_display")]
pub cosigned: BitVec,

...

fn ser_display<T, S>(value: &T, serializer: S) -> Result<S::Ok, S::Error>
where
    T: Display,
    S: Serializer
{
    serializer.collect_str(value)
}
```
for both fields?

Reply:
IMO representing an array of bits as a JSON array is more sensible than a string. :man_shrugging: